hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/api/StringMatcher.kt

```diff
@@ -25,19 +29,33 @@ data class StringMatcher(
   }
 
   override fun buildQuery(attributes: Attributes): AbstractQueryBuilder<*> =
-    attributes[attribute]?.let {
-      when (condition) {
-        StringCondition.IS -> QueryBuilders.termQuery(it.openSearchName, searchTerm).caseInsensitive(true)
-        StringCondition.IS_NOT -> QueryBuilders.boolQuery().mustNot(QueryBuilders.termQuery(it.openSearchName, searchTerm).caseInsensitive(true))
-        StringCondition.CONTAINS -> QueryBuilders.wildcardQuery(it.openSearchName, "*$searchTerm*").caseInsensitive(true)
-      }
-    } ?: throw AttributeSearchException("Attribute $attribute not recognised")
+    attributes[attribute]
+      ?.let { attr ->
+        val query = when (condition) {
+          IS -> QueryBuilders.termQuery(attr.openSearchName, searchTerm).caseInsensitive(true)
+          IS_NOT -> QueryBuilders.boolQuery().mustNot(QueryBuilders.termQuery(attr.openSearchName, searchTerm).caseInsensitive(true))
+          CONTAINS -> QueryBuilders.wildcardQuery(attr.openSearchName, "*$searchTerm*").caseInsensitive(true)
+        }
+        return when {
+          attr.isFuzzy && condition == IS -> {
+            QueryBuilders.boolQuery()
+              .should(query)
+              .should(QueryBuilders.fuzzyQuery(attr.openSearchName, searchTerm))
+          }
+          attr.isFuzzy && condition == CONTAINS -> {
+            QueryBuilders.boolQuery()
+              .should(query)
+              .should(QueryBuilders.matchQuery(attribute, searchTerm).fuzziness(Fuzziness.AUTO))
+          }
+          else -> query
+        }
```

Comment:
For fuzzy attributes we don't want to lose the standard behaviour, just add a fuzzy search to it.

Reply:
@mikehalmamoj does that mean for instance first name will be fuzzy in the this context if you like it or not? So if I search for "Marke" i will get "Marc" back as well and I can't control that behaviour?

Reply:
Yes that's correct. When we spoke to the product guys the other day they were adamant that it's better to receive too much data rather than miss something.

However it does feel like we should allow it to be turned off for other use cases - just don't have any yet. Do you think I should add a query parm to turn it on and off? What should the default be?